File: go/topjslibs/page_js_sources_test.go
package topjslibs

import (
	"testing"
)

func TestGet(t *testing.T) {
	// Arrange
	pageJsSources := PageJsSources{func() string {
		return getHtmlFrom("test.de.html")
	}}
	expected := []string{
		"base.min;v35620016.js",
		"modules.min;v35620016.js",
		"webtrekk_cookieControl.min;v35620016.js",
		"webtk.min;v35620016.js",
	}

	// Act
	actual := pageJsSources.get()
	actualMap := make(map[string]bool)
	for _, value := range actual {
		actualMap[value] = true
	}

	// Assert
	if len(expected) != len(actual) {
		t.Errorf("PageJsSources.get() = %s; want %s", actual, expected)
	}

	for _, expectedValue := range expected {
		if _, ok := actualMap[expectedValue]; !ok {
			t.Errorf("PageJsSources.get() = %s; want %s", actual, expected)
		}
	}
}

File: rust/src/top_js_libs/page_js_sources_test.rs
use std::collections::HashSet;

use crate::top_js_libs::page_js_sources::PageJsSources;
use crate::top_js_libs::test_util::get_html_from;

#[test]
fn get() {
    // Arrange
    let page_js_sources = PageJsSources::new(Box::from(|| get_html_from("test.de.html")));
    let mut expected = HashSet::new();
    expected.insert("base.min;v35620016.js".to_string());
    expected.insert("modules.min;v35620016.js".to_string());
    expected.insert("webtrekk_cookieControl.min;v35620016.js".to_string());
    expected.insert("webtk.min;v35620016.js".to_string());

    // Act
    let actual = page_js_sources.get();

    // Assert
    assert_eq!(expected, actual);
}